contracts/sui/contracts/sources/stork.move

```diff
@@ -0,0 +1,172 @@
+module stork::stork {
+
+    // === Imports ===
+
+    use stork::admin::AdminCap;
+    use stork::state::{Self, StorkState};
+    use stork::temporal_numeric_value_feed::{Self, TemporalNumericValueFeed};
+    use stork::temporal_numeric_value::TemporalNumericValue;
+    use stork::encoded_asset_id::{Self, EncodedAssetId};
+    use stork::verify::verify_stork_evm_signature;
+    use stork::event::Self;
+    use stork::update_temporal_numeric_value_evm_input::UpdateTemporalNumericValueEvmInput;
+    use stork::update_temporal_numeric_value_evm_input_vec::UpdateTemporalNumericValueEvmInputVec;
+    use sui::sui::SUI;
+    use sui::coin::Coin;
+    use sui::object_table::ObjectTable;
+
+    // === Errors ===
+
+    const EInvalidSignature: u64 = 0;
+    const EInsufficientFee: u64 = 1;
+    const EFeedNotFound: u64 = 2;
+
+    // === Entry Functions ===
+
+    entry fun init_stork(
+        // admin capability
+        _: &AdminCap,
+        // the address of the Stork program
+        stork_sui_public_key: address,
```

Comment:
Does SUI use a different native signing method than EVM? The intention for Solana is that we may one day want to include a Solana public key to use for signing instead of using the EVM default signing strategy. I'm not sure if we would want to do the same for SUI. If not, might be best to remove this.